libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -876,5 +874,47 @@ class kv_table {
       constexpr size_t num_elems = (sizeof(U) / sizeof(kv_index)) - 1;
       for_each_field<num_elems>(u, f);
    }
+
+   template <typename V>
+   static void serialize(const V& value, void* buffer, size_t size) {
+      datastream<char*> ds((char*)buffer, size);
+      unsigned_int i{0};
+      ds << i;
+      ds << value;
+   }
+
+   template <typename... Vs>
+   static void serialize(const std::variant<Vs...>& value, void* buffer, size_t size) {
+      datastream<char*> ds((char*)buffer, size);
+      ds << value;
+   }
+
+   template <typename V>
+   static void deserialize(V& value, void* buffer, size_t size) {
+      unsigned_int idx;
+      datastream<const char*> ds((char*)buffer, size);
+
+      ds >> idx;
+      eosio::check(idx==unsigned_int(0), "there was an error deserializing this value.");
+      ds >> value;
+   }
+
+   template <typename... Vs>
+   static void deserialize(std::variant<Vs...>& value, void* buffer, size_t size) {
+      datastream<const char*> ds((char*)buffer, size);
+      ds >> value;
+   }
+
+   template <typename V>
+   static size_t get_size(V&& value) {
```

Comment:
You should also change this overload to prevent errors.  I don't see any reason for get_size to take an rvalue.